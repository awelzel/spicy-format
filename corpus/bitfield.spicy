module foo;

type X = unit {
	x: bitfield(8) {
		a: 0..1; # foo
		b: 0..8; # bar
	};
};

type X =   bitfield(8) {
    a: 0..3=2;
    b: 4..7;
    c: 7     = 1;
  };

type X = unit {
	:bitfield(8) { a: 0..1; b: 0..7; } &bit-order=spicy::BitOrder::MSB0;
	:bitfield(8) { c: 0..1; d: 0..7; };
	:bitfield(8) { e: 0..1; f: 0..7; };
};
