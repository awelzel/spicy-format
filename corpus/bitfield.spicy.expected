module foo;

type X = unit {
    x: bitfield(8) {
        a: 0..1; # foo
        b: 0..8; # bar
    };
};
